src/sdk/object_sdk.js

```diff
@@ -95,6 +95,15 @@ class ObjectSDK {
         return bucket.namespace;
     }
 
+    async read_bucket_sdk_caching_info(name) {
+        try {
+            const { bucket } = await bucket_namespace_cache.get_with_cache({ sdk: this, name });
+            return bucket.namespace.caching;
```

Comment:
This crashes if we get here with a regular non-namespace bucket.
Since you are calling that on every request in s3_rest, you will just crash the S3 endpoint and all requests on non-namespaced buckets won't work.

Reply:
fixed